starlette_jsonapi/resource.py

```diff
@@ -130,17 +134,24 @@ async def validate_body(self, partial=None) -> dict:
             raise JSONAPIException(status_code=400, errors=errors.get('errors'))
         return body
 
-    async def serialize(self, data: Any, many=False, *args, **kwargs) -> dict:
+    async def serialize(self, data: Any, many=False, paginate=False, *args, **kwargs) -> dict:
         """
         Serializes data as a JSON:API payload and returns a `dict`
         which can be passed when calling `BaseResource.to_response`.
 
         Additional args and kwargs are passed to the `marshmallow` based Schema.
         """
+        links = None
+        if self.pagination_class and paginate:
```

Comment:
I think we should check `paginate` only, and let `paginate_request` handle the case where `pagination_class` is not defined by throwing an exception. 
Otherwise, a developer might forget they passed paginate=True but didn't configure the pagination class on their resource.